pkgs/applications/misc/mepo/default.nix

```diff
@@ -16,26 +16,53 @@
 , xdotool
 , bemenu
 , withX11 ? false
+
+# transmits user's location without asking first
+, geoclueSupport ? false , geoclue2
+
+# transmits user's location without asking first
+, reportLocationToMozilla ? false
+
+, gpsdSupport ? true , gpsd
 }:
 
 let
   menuInputs = if withX11 then [ dmenu xdotool ] else [ bemenu ];
 in stdenv.mkDerivation rec {
   pname = "mepo";
-  version = "0.4.2";
+  version = "0.5";
 
   src = fetchFromSourcehut {
     owner = "~mil";
     repo = pname;
     rev = version;
-    hash = "sha256-k6YXaqB3EwbDPlTvijZf10q+IYwt4/MiqGXL495KIcY=";
+    hash = "sha256-V8NdlDnj6nyC7pAPu76tMbw//5LqgPcd0jOKSsvx9ZU=";
   };
 
   nativeBuildInputs = [ pkg-config zig makeWrapper ];
 
   buildInputs = [
     curl SDL2 SDL2_gfx SDL2_image SDL2_ttf inconsolata-nerdfont jq ncurses
-  ] ++ menuInputs;
+  ] ++ menuInputs
+    ++ lib.optional geoclueSupport geoclue2
+    ++ lib.optional gpsdSupport gpsd;
+
+  patches = lib.optionals (!reportLocationToMozilla) [
+    ./dont-report-user-location-to-mozilla-without-asking.patch
+  ] ++ lib.optionals (!geoclueSupport) [
+    ./dont-report-user-location-to-geoclue-without-asking.patch
+  ];
```

Comment:
conditional patches are not great. Since this is just shell I would suggest to make this conditional based on an ENV or so and set that one for a wrapper.

Reply:
Good point; I will revise.

Reply:
On second thought: inventing our own environment variable for this is a bad idea.  When/if upstream gets around to responding to the bug filed about this being a bad idea for default behavior, they will create their own mechanism for opting in/out of this.  Then we'd either have to support the nixpkgs-specific environment variable forever (yuck) or have a breaking change with a really bad silent failure behavior.

So, I don't think inventing an environment variable is a great idea.

I think the only option for deconditionalizing the patches is to remove the options and apply the patches unconditionally.  This preserves the 0.4 behavior (never report the user's location to third parties).

I've implemented this in 926f3e6429a7.

Reply:
> conditional patches are not great

What possible issues may conditional patching cause?

Reply:
> What possible issues may conditional patching cause?

If the patch is conditional and most users aren't building with that condition, nobody will notice if the patch ceases to apply cleanly.  For example, when the version is bumped.

So in general it is preferable to have any patches applied unconditionally -- when it's possible to do that.